### ISSUE TITLE
Proxy based patch generation

### ISSUE THREAD

**Issue by Korijn** [fork-tongue/patchdiff#8]
As an alternative to diffing, one can also monitor mutations as they are being made against the input and emit patches, by using a proxy mechanism. This is how [Immer](https://immerjs.github.io/immer/) works. It's much more efficient than diffing, of course.

**Comment by berendkleinhaneveld** [comment]
Yeah, I wanted to propose to do something like this within observ actually as we have all the proxy stuff in place.

**Comment by Korijn** [comment]
I thought about doing that, but I think it requires very different trap implementations and assignments, so we might as well implement it here in this library and keep observ focused on what it does best.

**Comment by Korijn** [comment]
The magic is here: https://github.com/immerjs/immer/blob/master/src/core/proxy.ts
